tests/lib/rules/yoda.js

```diff
@@ -992,6 +1017,110 @@ ruleTester.run("yoda", rule, {
                     type: "BinaryExpression"
                 }
             ]
+        },
+        {
+            code: "if (`green` < x.y && x.y < `blue`) {}",
+            output: "if (`green` < x.y && `blue` > x.y) {}",
+            options: ["always", { exceptRange: true }],
+            parserOptions: { ecmaVersion: 2015 },
+            errors: [
+                {
+                    messageId: "expected",
+                    data: { expectedSide: "left", operator: "<" },
+                    type: "BinaryExpression"
+                }
+            ]
+        },
+        {
+            code: "if (x < -1n || 1n <= x) {}",
+            output: "if (x < -1n || x >= 1n) {}",
+            options: ["never", { exceptRange: true }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [
```

Comment:
I think cases like these should be warned.
There are also other false-positive cases that use the operator `||`.
I think the `isOutsideTest()` didn't check it properly.

Reply:
> `(x < -1n || 1n <= x)`

this is not a `yoda` and I think `never` with `exceptRange` only accepts `Yoda` for range.

Reply:
fixed